### ISSUE TITLE
Tensorflow 2 versions not training. Stuck at epoch 1.

### ISSUE THREAD

**Issue by sachinkmohan** [matterport/Mask_RCNN#2808]
This is a wonderful repository! I am trying to apply Tensorflow model optimization(pruning and quantization), I was able to do that. But stuck at training.
I have really hard time get this network trained on TF versions > 2.0
The training gets stuck in the first epoch as mentioned by a lot of people in many of the issues raised here.

As mentioned by one of the comments, I am using https://github.com/leekunhee/Mask_RCNN repository converted for TF2. Everything works until training.

One solution was to ensure if I have enabled GPU, and yes it is enabled.
```
print("Num GPUs Available: ", len(tf.config.experimental.list_physical_devices('GPU')))

Num GPUs Available: 1
```

Below are settings maintained in `requirements.txt`
```
Pillow
cython
matplotlib
scikit-image
tensorflow==2.2.0
keras==2.3.1
opencv-python
h5py
imgaug
IPython[all]
easydict
```
Please help

**Comment by valavanisleonidas** [comment]
Hello, I think the problem is with the multiprocess in TF > 2.0. So in this repo [here](https://github.com/sabderra/Mask_RCNN/blob/850944b385c1a287028abade463262357ef5bff2/mrcnn/model.py#L2388) the multiprocessing is False which allows for training. I have tested this repo myself and seems to work.

I think training stucks if Multiprocess is True

Hope it helps

**Comment by Kamlesh364** [comment]
you can use [this](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.8.0)  one, for TF==2.7.0

**Comment by sachinkmohan** [comment]
Ok thanks @valavanisleonidas and @Kamlesh364 for your help! I will have them checked! :)

**Comment by avnerst** [comment]
> Hello, I think the problem is with the multiprocess in TF > 2.0. So in this repo [here](https://github.com/sabderra/Mask_RCNN/blob/850944b385c1a287028abade463262357ef5bff2/mrcnn/model.py#L2388) the multiprocessing is False which allows for training. I have tested this repo myself and seems to work.
> 
> I think training stucks if Multiprocess is True
> 
> Hope it helps

thanks a lot first of all!
so I'm running the training code on a Linux VM, and after using this version to update mrcnn to support TensorFlow 2, and run it on GPU, it doesn't save all the weights after each iteration.
when running the code first with a small training set size (4 images) for 5 epochs, all the weights get saved except for epoch 4 for some reason.
when running it for 10 epochs using a larger training set size of 700 images, it only saves the weights for epochs 1 and 2 and then stops (sometimes saving the weights only for epochs 1 and 3).
did anybody experience this or know how to fix it?
thanks!
